### PR TITLE
Fix potential github action smells

### LINKED ISSUE #2411: Fix potential github action smells
### Describe the bug

Hey! 🙂
During my research I've found the following potential smells: 

 - Avoid executing  scheduled workflows on forks
 - Use names for run steps
 - Define permissions for workflows with external actions
 - Steps should only perform a single command
 - Use commit hash instead of tags for action versions

I'll fix them in a PR and would love to hear your feedback!  

 These changes are part of a research Study at TU Delft looking at GitHub Action Smells. [Find out more](https://ceddy4395.github.io/research/gha-smells.html)

### Reproduction

_No response_

### Expected behavior

_No response_

### Platform and versions

```Shell
Github actions
```

### Stack trace

_No response_

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by ceddy4395**
Hey! 🙂
I want to contribute the following changes to your workflow:

 - Avoid executing  scheduled workflows on forks
 - Use names for run steps
 - Define permissions for workflows with external actions
 - Steps should only perform a single command
 - Use commit hash instead of tags for action versions
 
 These changes are part of a research Study at TU Delft looking at GitHub Action Smells. [Find out more](https://ceddy4395.github.io/research/gha-smells.html)

Closes #2411

### CODE REVIEW

File: .github/workflows/cache-factory.yaml [hunk 1]
```diff
@@ -65,13 +68,17 @@ jobs:
...
       - name: Checkout repository
-        uses: actions/checkout@v4
+        uses: actions/checkout@b4ffde6  # v4
 
       - name: Symlink target to C:\
         if: ${{ runner.os == 'Windows' }}
         shell: powershell
         run: |
           New-Item -ItemType Directory -Force -Path C:\spacedrive_target
+      - name: Symlint target to C:\
```
Comment (utkubakir) [inline .github/workflows/cache-factory.yaml:78]: `symlint`

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -20,10 +20,12 @@ concurrency:
 jobs:
   typescript:
     name: TypeScript
-    runs-on: ubuntu-latest
+    runs-on: ubuntu-22.04
+    timeout-minutes: 3
```
Comment (utkubakir) [inline .github/workflows/ci.yml:24]: is 3 minutes too low?

File: .github/workflows/ci.yml [hunk 3]
```diff
@@ -117,12 +127,17 @@ jobs:
         shell: powershell
         run: |
           New-Item -ItemType Directory -Force -Path C:\spacedrive_target
+      - name: Symlint target to C:\
```
Comment (utkubakir) [inline .github/workflows/ci.yml:130]: `symlint`

**Review by utkubakir — CHANGES_REQUESTED** [review]

File: .github/workflows/ci.yml (hunk 2 above)
Comment (utkubakir) [reply .github/workflows/ci.yml:24]: ok looked through some of our previous jobs and we seem to be always under 3 min so this is good.

File: .github/workflows/cache-factory.yaml (hunk 1 above)
Comment (ceddy4395) [reply .github/workflows/cache-factory.yaml]: fixed :)

File: .github/workflows/ci.yml (hunk 3 above)
Comment (ceddy4395) [reply .github/workflows/ci.yml]: Fixed :)

File: .github/workflows/cache-factory.yaml [hunk 4]
```diff
@@ -65,13 +68,17 @@ jobs:
...
       - name: Checkout repository
-        uses: actions/checkout@v4
+        uses: actions/checkout@1e31de5234b9f8995739874a8ce0492dc87873e2  # v4
 
       - name: Symlink target to C:\
         if: ${{ runner.os == 'Windows' }}
         shell: powershell
         run: |
           New-Item -ItemType Directory -Force -Path C:\spacedrive_target
+      - name: symlint target to C:\
```
Comment (utkubakir) [inline .github/workflows/cache-factory.yaml:78]: woops, I was actually trying to highlight the typo, it should be `Symlink`, sorry...

Comment (ceddy4395) [reply .github/workflows/cache-factory.yaml]: Aaah I see. Updated!

**Review by utkubakir — APPROVED** [review]
lgtm, thank you

**Pull request merged**